lib/matplotlib/tests/test_contour.py

```diff
@@ -221,6 +221,23 @@ def test_log_locator_levels(split_collections):
     _maybe_split_collections(split_collections)
 
 
+@pytest.mark.parametrize("n_levels", [2, 3, 4, 5, 6])
```

Comment:
FWIW, only the `n_levels=2` case fails on `main`. 

On main, the `levels` argument doesn't seem to do much of anything, there are always 4 `visible_levels`.

On the PR branch it still is a bit odd, there are only ever 2 (for levels= 2 or 3) or 4 (for levels>=4)

Is this expected?

Reply:
I've updated the top comment on this PR with a manual testing script. On `main` the levels are always 7, presumably to avoid too many levels? If requesting `levels=20` and only getting 7 is not desired it seems orthogonal to this fix.

I wonder if the number of levels shown is related to the data being contoured? I've updated the test data to be the same as the example above so the [2, 3, 4] cases fail on main.